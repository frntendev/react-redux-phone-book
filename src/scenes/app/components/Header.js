import React from "react";
import { NavLink } from "react-router-dom";
import { css } from "react-emotion";

const styles = {
  header: css`
    background: #833471;
    height: 50px;
    box-shadow: 0px 1px 5px 0px rgba(0, 0, 0, 0.2),
      0px 2px 2px 0px rgba(0, 0, 0, 0.14), 0px 3px 1px -2px rgba(0, 0, 0, 0.12);
    position: fixed;
    width: 100%;
    z-index: 9;
    label: header;
  `,
  nav: css`
    max-width: 960px;
    height: inherit;
    margin: 0 auto;
    display: grid;
    grid-template-columns: 100px 150px 150px;
    grid-gap: 5px;
    align-items: center;
    label: nav;
    @media (max-width: 768px) {
      grid-template-columns: 1fr 1fr;
    }
  `,
  link: css`
    height: inherit;
    line-height: 50px;
    text-align: center;
    text-decoration: none;
    color: #fff;
    font-size: 1.3rem;
    background: #6f1e51;
    transition: 0.3s all ease;
    position: relative;
    overflow: hidden;
    label: link;
    &:hover {
      background: #4a1135;
    }
  `,
  badge: css`
    min-width: 20px;
    height: 20px;
    position: absolute;
    top: 5px;
    right: 8px;
    border-radius: 10px;
    background: #ffda26;
    font-size: 0.8rem;
    display: flex;
    justify-content: center;
    align-items: center;
    color: #1c80d0;
    transition: 0.3s all ease;
    label: badge;
  `,
  adding: css`
    background: #65ff26;
    transform: scale(1.2);
  `,
  active: css`
    && {
      background: #4a1135;
    }
  `
};
const Header = () => {
  return (
    <header className={styles.header}>
      <nav className={styles.nav}>
        <NavLink
          activeClassName={styles.active}
          exact
          className={styles.link}
          to="/"
        >
          Home
        </NavLink>
        <NavLink
          activeClassName={styles.active}
          exact
          className={styles.link}
          to="/blacklist"
        >
          Black List
        </NavLink>
      </nav>
    </header>
  );
};

export default Header;
