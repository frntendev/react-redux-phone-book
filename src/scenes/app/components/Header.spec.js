import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Header from "./Header";
configure({ adapter: new Adapter() });

const setup = props => {
  const component = shallow(<Header />);

  return {
    component: component
  };
};

describe("Header component", () => {
  it("should render successfully", () => {
    const { component } = setup();
    expect(component).toMatchSnapshot();
  });
});
