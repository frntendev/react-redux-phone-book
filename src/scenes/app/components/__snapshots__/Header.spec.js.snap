// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render successfully 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <nav
        className="css-1kededo-nav"
      >
        <NavLink
          activeClassName="css-vvradc"
          ariaCurrent="true"
          className="css-71tcxc-link"
          exact={true}
          to="/"
        >
          Home
        </NavLink>
        <NavLink
          activeClassName="css-vvradc"
          ariaCurrent="true"
          className="css-71tcxc-link"
          exact={true}
          to="/blacklist"
        >
          Black List
        </NavLink>
      </nav>,
      "className": "css-94njij-header",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="css-vvradc"
            ariaCurrent="true"
            className="css-71tcxc-link"
            exact={true}
            to="/"
          >
            Home
          </NavLink>,
          <NavLink
            activeClassName="css-vvradc"
            ariaCurrent="true"
            className="css-71tcxc-link"
            exact={true}
            to="/blacklist"
          >
            Black List
          </NavLink>,
        ],
        "className": "css-1kededo-nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "css-vvradc",
            "ariaCurrent": "true",
            "children": "Home",
            "className": "css-71tcxc-link",
            "exact": true,
            "to": "/",
          },
          "ref": null,
          "rendered": "Home",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "css-vvradc",
            "ariaCurrent": "true",
            "children": "Black List",
            "className": "css-71tcxc-link",
            "exact": true,
            "to": "/blacklist",
          },
          "ref": null,
          "rendered": "Black List",
          "type": [Function],
        },
      ],
      "type": "nav",
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <nav
          className="css-1kededo-nav"
        >
          <NavLink
            activeClassName="css-vvradc"
            ariaCurrent="true"
            className="css-71tcxc-link"
            exact={true}
            to="/"
          >
            Home
          </NavLink>
          <NavLink
            activeClassName="css-vvradc"
            ariaCurrent="true"
            className="css-71tcxc-link"
            exact={true}
            to="/blacklist"
          >
            Black List
          </NavLink>
        </nav>,
        "className": "css-94njij-header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="css-vvradc"
              ariaCurrent="true"
              className="css-71tcxc-link"
              exact={true}
              to="/"
            >
              Home
            </NavLink>,
            <NavLink
              activeClassName="css-vvradc"
              ariaCurrent="true"
              className="css-71tcxc-link"
              exact={true}
              to="/blacklist"
            >
              Black List
            </NavLink>,
          ],
          "className": "css-1kededo-nav",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "css-vvradc",
              "ariaCurrent": "true",
              "children": "Home",
              "className": "css-71tcxc-link",
              "exact": true,
              "to": "/",
            },
            "ref": null,
            "rendered": "Home",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "css-vvradc",
              "ariaCurrent": "true",
              "children": "Black List",
              "className": "css-71tcxc-link",
              "exact": true,
              "to": "/blacklist",
            },
            "ref": null,
            "rendered": "Black List",
            "type": [Function],
          },
        ],
        "type": "nav",
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
